src/vm.rs

```diff
@@ -414,34 +454,75 @@ impl<'a, S: Storage, C: PevmChain> Database for VmDb<'a, S, C> {
     }
 
     fn storage(&mut self, address: Address, index: U256) -> Result<U256, Self::Error> {
+        if let LazyStrategy::ERC20Transfer {
+            sender_balance_slot,
+            recipient_balance_slot,
+        } = self.lazy_strategy
+        {
+            if index == sender_balance_slot {
+                return self
```

Comment:
Can we mock `U256::MAX` here to save a storage read?

Reply:
Short answer: no.

Long answer: It's a bit hard to explain. The main idea is to perserve the zeroness of the storage slots.

Imagine Alice sends to Bob 10 tokens. Naturally, we think about `Alice -= 10; Bob += 10;`.
However, the following initial conditions must be treated differently:

1. Alice = 10. Bob = 0.
2. Alice = 11. Bob = 0.
3. Alice = 10. Bob = 1.
4. Alice = 11. Bob = 1.

Look at the zeroness, we have:

1. Alice: some -> none. Bob: none -> some.
2. Alice: some -> some. Bob: none -> some.
3. Alice: some -> none. Bob: some -> some.
4. Alice: some -> some. Bob: some -> some.

The zeroness affects the gas used.

We must choose one among 4 cases. If we choose wrongly, it is _very hard_ to fix. One might think about adjusting the beneficiary rewards in post. However, it's much more complicated.

Currently, at the end, if it is detected that we messed up with the zeroness, then it will fallback to sequential :(

For now, I claim that using storage value gives us the best prediction. At least for the current blocks, this strategy succeeds 99.99% of the times.